torchmultimodal/modules/encoders/albef_vision_encoder.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from functools import partial
+from typing import Callable
+
+import torch
+from torch import nn
+from torchvision.models.vision_transformer import VisionTransformer
+
+
+class ALBEFVisionEncoder(VisionTransformer):
+    def __init__(
+        self,
+        image_size: int = 256,
+        patch_size: int = 16,
+        num_layers: int = 12,
+        num_heads: int = 12,
+        hidden_dim: int = 768,
+        mlp_dim: int = 3072,
+        num_classes: int = 768,
+        norm_layer: Callable[..., torch.nn.Module] = partial(nn.LayerNorm, eps=1e-6),
+    ):
+        super().__init__(
+            image_size=image_size,
+            patch_size=patch_size,
+            num_layers=num_layers,
+            num_heads=num_heads,
+            hidden_dim=hidden_dim,
+            mlp_dim=mlp_dim,
+            num_classes=num_classes,
+            norm_layer=norm_layer,
+        )
+        self.heads = None
+
+    def forward(self, x: torch.Tensor):
+        # Reshape and permute the input tensor
+        x = self._process_input(x)
+        n = x.shape[0]
+
+        # Expand the class token to the full batch
+        batch_class_token = self.class_token.expand(n, -1, -1)
+        x = torch.cat([batch_class_token, x], dim=1)
+
+        x = self.encoder(x)
```

Comment:
Why not just import the encoder from torchvision and set that as self.encoder in the init? Then stitch together with the conv layer used in _process_input, and you should be able to remove the use of inheritance altogether